main.c

```diff
@@ -819,9 +819,71 @@ resolve_all_solvables (Pool  *pool,
   return solv_failed;
 }
 
+static void
+add_solvable_to_pile (const char *solvable,
+                      Pool       *pool,
+                      Queue      *selection,
```

Comment:
If you create the selection inside this function, you don't have to pass it to both `add_solvable_to_pile` and `add_solvables_from_file_to_pile` (where it's not even used other than to pass it down to `add_solvable_to_pile`).

Reply:
Nice catch. I'll take care of that.